### PR TITLE
Basic fix for Images' "No Results + filter" message

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Users using image search

Relates to #10287 
Closes #10325

## What is it doing for them?
A few of us worked on this fix ([see original PR](https://github.com/wellcomecollection/wellcomecollection.org/pull/10325)), but with the end to ends being flaky (fixed in other PRs 🙌 ) and it being an older PR with a lot of investigative code changes, I thought it'd be good to start from scratch with the knowledge we now have.

What fixes the issues;
- Manually pass in the dates filters to `hasActiveFilters` so it knows the consider them (see comment in code)
- Modal filters only rendered when the modal is active ([see this thread in the other PR that explains why it needs doing](https://github.com/wellcomecollection/wellcomecollection.org/pull/10325#discussion_r1430367321))

Other work done;
- Refactoring of `ResetActiveFilters` based on [Jamie's work in the previous PR](https://github.com/wellcomecollection/wellcomecollection.org/commit/cada0442dd857e5d6670d96c9d248a47d2dd4ff0)
- Refactor of `hasFilter` based on [Jamie's work in the previous PR](https://github.com/wellcomecollection/wellcomecollection.org/commit/3409002d08cec7e7aa6cef5aa3227f2f3855b922)
- Make `hasNoResults` a transient prop in the `Wrapper` styled component so it's not passed to the DOM

### CODE REVIEW

**Comment by jamieparkinson** [comment]
I think it might be useful to replicate the refactors from https://github.com/wellcomecollection/wellcomecollection.org/commit/3409002d08cec7e7aa6cef5aa3227f2f3855b922 and particularly from https://github.com/wellcomecollection/wellcomecollection.org/commit/cada0442dd857e5d6670d96c9d248a47d2dd4ff0 - the latter is a mess and I think the refactor fixed some weird behaviours

**Comment by rcantin-w** [comment]
> I think it might be useful to replicate the refactors from https://github.com/wellcomecollection/wellcomecollection.org/commit/3409002d08cec7e7aa6cef5aa3227f2f3855b922 and particularly from https://github.com/wellcomecollection/wellcomecollection.org/commit/cada0442dd857e5d6670d96c9d248a47d2dd4ff0 - the latter is a mess and I think the refactor fixed some weird behaviours

Yes I was going to ask if anything else should be moved in, but I want to run tests one by one. We were getting unwanted behaviour (the colour filter always applied?) and I wasn't sure where it came from. I'll look at adding those back in if all good. In the meantime I think these basic changes do fix the og bug! 🤞

**Comment by georgiaewhitney** [comment]
I can't replicate the bug in this branch! 🥳

**Comment by rcantin-w** [comment]
Nice, thanks for checking! For both @georgiaewhitney and @jamieparkinson I've now edited the description and made the PR ready for review.

File: content/webapp/pages/search/works.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const SortPaginationWrapper = styled.div`
   flex-wrap: wrap;
 `;
 
-export const CatalogueSearchPage: NextPageWithLayout<Props> = ({
+const CatalogueSearchPage: NextPageWithLayout<Props> = ({
```
Comment (rcantin-w) [inline content/webapp/pages/search/works.tsx:60]: No need to export here, it's exported at the bottom of the file in line with the other pages.

**Review by georgiaewhitney — COMMENTED** [review]
🎉

**Review by georgiaewhitney — APPROVED** [review]

**Pull request merged**